apps/src/blockly/addons/cdoXml.js

```diff
@@ -88,6 +93,34 @@ export default function initializeBlocklyXml(blocklyWrapper) {
 
   blocklyWrapper.Xml.createBlockOrderMap = createBlockOrderMap;
 }
+/**
+ * Gets the XML representation for a project, including its workspace and, if applicable, the hidden definition workspace.
+ *
+ * @param {Blockly.Workspace} workspace - The workspace from which to obtain the project XML.
+ * @returns {string} The XML representation of the project.
+ *
+ */
+export function getProjectXml(workspace) {
+  // Start by getting the XML for all blocks on the workspace.
+  const workspaceXml = Blockly.Xml.blockSpaceToDom(workspace);
+
+  if (shouldSkipHiddenWorkspace(workspace)) {
+    return workspaceXml;
+  }
+
+  // Also serialize blocks on the hidden workspace for procedure definitions.
+  const hiddenWsXml = Blockly.Xml.blockSpaceToDom(
```

Comment:
super nit: Even though it's longer, I think keeping `hiddenWorkspaceXml` is clearer